sdk/digitaltwins/Azure.DigitalTwins.Core/src/Queries/QueryBuilders/AdtQuerySelectFrom.cs

```diff
@@ -0,0 +1,108 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Azure.DigitalTwins.Core.QueryBuilder
+{
+    /// <summary>
+    /// Query that already contains a SELECT and FROM clause.
+    /// </summary>
+    internal class AdtQuerySelectFrom : AdtQuery
+    {
+        /// <summary>
+        /// Adds a WHERE and its conditional argument(s) clause to the query object. Meant to be used for simple
+        /// conditions involving operators or with basic ADT functions. Multiple WHERE clauses are appended using
+        /// the AND logical operator.
+        /// </summary>
+        /// <param name="condition"> A custom object that encodes the logical statement nested within the WHERE clause. </param>
+        /// <returns> ADT query that already contains SELECT and FROM. </returns>
+        public AdtQuerySelectFrom Where(BaseCondition condition)
```

Comment:
so did you decide that you will expose this option to your users?

Reply:
Yes. For now, even though BaseCondition is empty, I think it should stay in the hopes that a final decision will become more clear during implementation. ATM I think this method should be visible to users -- however, conditions can be reworked such that BaseCondition is no longer needed and this method could be replaced by two overloaded Where methods that each take a different kind of condition instead of the more generic option like it does now.